eth/filters/filter_system.go

```diff
@@ -174,6 +174,17 @@ func (sub *Subscription) Unsubscribe() {
 		// this ensures that the manager won't use the event channel which
 		// will probably be closed by the client asap after this method returns.
 		<-sub.Err()
+
+	drainLoop:
```

Comment:
I'm not sure it's needed. After the uninstall request gets processed, no more following events will be sent to this filter. All accumulated events will be consumed by the filter handler itself.

But anwway I agree it's ugly to prevent the deadlock in the Unsubscribe by such approach.

Reply:
The uninstall request is processed in `eventLoop`,  while `eventLoop` is busy pushing hash into  `sub.f.hashes`, however if  `sub.f.hashes` have unread `hash`, then deadlock happens.